### PR TITLE
Add `executeAsync` method

### PR DESCRIPTION
**Pull request by tkrs**
Added new method for asynchronous executions. And remove unneccessary `implicit val` in the `Try` object.

### CODE REVIEW

File: core/src/main/scala/agni/Functions.scala [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
+import com.google.common.util.concurrent.{ FutureCallback, Futures }
+
+import scala.collection.JavaConverters._
 
 trait Functions[F[_]] {
   type Action[A] = Kleisli[F, Session, A]
   def withSession[A](f: Session => F[A]): Action[A] = Kleisli(f)
+
+  implicit class ResultSetFuture0(f: ResultSetFuture) {
+    def callback[PI[_], A](p: PI[A], fa: (PI[A], Throwable) => Unit, fb: (PI[A], Iterator[A]) => Unit)(implicit decoder: RowDecoder[A]): PI[A] = {
```
Comment (yanana) [inline core/src/main/scala/agni/Functions.scala]: What is `PI` for?

File: core/src/main/scala/agni/std/Future.scala [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
-import scala.concurrent.{ ExecutionContext, Future => SFuture }
+import scala.concurrent.{ Future => SFuture, Promise }
 
 object Future extends Agni[SFuture, Throwable] {
-  implicit def scalaFutureMonadError(implicit EC: ExecutionContext) = MonadError[SFuture, Throwable]
+  type P[A] = Promise[Iterator[A]]
+  def executeAsync[A](query: Statement)(implicit decoder: RowDecoder[A]): Action[Iterator[A]] =
+    withSession { session =>
+      session.executeAsync(query).callback[P, A](
+        Promise(), _ failure _, _ success _
```
Comment (yanana) [inline core/src/main/scala/agni/std/Future.scala]: This style (`_ op _`) should be used only for binary operators, I think.

File: core/src/main/scala/agni/Functions.scala [hunk 3]
```diff
@@ -1,9 +1,24 @@
...
+      Futures.addCallback(f, new FutureCallback[ResultSet] {
```
Comment (yanana) [inline core/src/main/scala/agni/Functions.scala:21]: Is using `directExecutor` intended?

File: core/src/main/scala/agni/std/Future.scala (hunk 2 above)
Comment (tkrs) [inline core/src/main/scala/agni/std/Future.scala]: I see.

File: core/src/main/scala/agni/Functions.scala (hunk 1 above)
Comment (tkrs) [inline core/src/main/scala/agni/Functions.scala]: I have no reason... use `G[_]` instead.

File: core/src/main/scala/agni/Functions.scala (hunk 3 above)
Comment (tkrs) [inline core/src/main/scala/agni/Functions.scala:21]: No. I was wrong. `Executor` was wont to be customizable.

**Comment by yanana** [comment]
LGTM

**Pull request merged**